File: lec_02/my_benchmark/hexstring.ts
var x: bigint = 0x0012345fedcba077n;
intToHex(x);

function intToHex(x: bigint) {
  //var hex = new String("0123456789abcdef"); 
  //var str = new Array(0);

  for (let i = 15; i >= 0n; i = i - 1n) {
    x = x * 16;
  }

  console.log(x);
}

File: bril_intro/lec_02/hexstring/hexstring.py
# POC

a = 0x12345fedcba00077

hex = '0123456789abcdef'

s = [''] * 16

acc = a

R = list(range(0,16))
R.reverse()

for d in R:
  s[d] = hex[acc & 0xf]
  acc = acc >> 4

print("0x%x" % a)
print(''.join(s))
